alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionEvaluator.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.alerter;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.hawkular.metrics.alerter.ConditionExpression.Function;
+import org.hawkular.metrics.model.AvailabilityBucketPoint;
+import org.hawkular.metrics.model.BucketPoint;
+import org.hawkular.metrics.model.NumericBucketPoint;
+import org.hawkular.metrics.model.Percentile;
+import org.jboss.logging.Logger;
+
+import com.udojava.evalex.Expression;
+import com.udojava.evalex.Expression.ExpressionException;
+
+public class ConditionEvaluator {
+    private static final Logger log = Logger.getLogger(ConditionEvaluator.class);
+
+    // Map variableName:query
+    private Map<String, QueryFunc> queryVars;
+
+    private String eval;
+    private Expression expression;
+
+    public ConditionEvaluator(String eval) {
+        super();
+
+        this.eval = eval;
+        this.queryVars = new HashMap<>();
+
+        this.expression = new Expression(replaceQueriesWithVariables(eval));
+
+        log.debugf("eval [%s] produced [%s] with variables %s", eval, expression.getOriginalExpression(), queryVars);
+
+        // Do a test evaluation to validate the expression
+        try {
+            for (String var : queryVars.keySet()) {
+                this.expression.setVariable(var, "1");
+            }
+            this.expression.eval();
+        } catch (Exception e) {
+            throw new ExpressionException("Invalid eval [" + eval + "]: " + e.getMessage());
```

Comment:
Perhaps a IllegalArgumentException instead to propagate this one com.udojava.evalex.Expression.ExpressionException ?